include/rocksdb/utilities/backupable_db.h

```diff
@@ -165,6 +163,34 @@ struct BackupableDBOptions {
   // db_session_id as a fallback.
   BackupTableNameOption share_files_with_checksum_naming;
 
+  // Option for custom checksum functions.
+  // When this option is nullptr, BackupEngine will use its default crc32c as
+  // the checksum function.
+  //
+  // When it is not nullptr, BackupEngine will try to find in the factory the
+  // checksum function that DB used to calculate the file checksums. If such a
+  // function is found, BackupEngine will use it to create, verify, or restore
+  // backups, in addition to the default crc32c checksum function. Therefore,
+  // this option comes into effect only if DB has a custom checksum factory and
+  // this option is set to the same factory.
+  //
+  // Regardless of what is set for this option, BackupEngine will always use
+  // its default crc32c checksum for creating, verifying, or restoring backups.
+  //
+  // When the DB checksum factory is set to GetFileChecksumGenCrc32cFactory(),
+  // we recommend not setting this option to the same factory to avoid
+  // unnecessary additional computation of crc32c since the backup engine
+  // default checksum function is also crc32c.
+  //
+  // Note: If share_files_with_checksum and share_table_files are true,
+  // the <checksum> appeared in the table filenames will be the custom checksum
+  // value if db session ids are available (namely, table file naming options
+  // is kOptionalChecksumAndDbSessionId and the db session ids obtained from
+  // the table files are nonempty).
+  //
+  // Default: nullptr
```

Comment:
I think it's worth adding another note here that we do not require the same setting to file_checksum_gen_factory during restore as was set during backup (each can do the extra checksum verification independently vs. db manifest), but we strongly recommend setting it to the same as the DB file checksum function for all BackupEngine interactions when practical.